### PR TITLE
Change: switch from flask_script to click, add CLI unit tests and upgrade Flask version

### LINKED ISSUE #1112: Upgrade Flask to 0.11.1
- [ ] Upgrade all Flask extensions
- [ ] Upgrade to latest Flask 0.11 release
- [ ] Bonus: replace Flask-Script with Click

### PR DESCRIPTION
**Pull request by washort**
This provides basic test coverage for the CLI commands and replaces `flask_script`.

Closes #1112.

### CODE REVIEW

File: manage.py [hunk 1]
```diff
@@ -2,62 +2,75 @@
 """
 CLI to manage redash.
 """
+import code
```
Comment (arikfr) [inline manage.py]: Seems unused?

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+
+    def test_password_bad_org(self):
+        runner = CliRunner()
+        result = runner.invoke(
+            password, ['foobar@example.com', 'xyzzy', '--org', 'default'])
+        self.assertTrue(result.exception)
+        self.assertEqual(result.exit_code, 1)
+        self.assertIn('not found', result.output)
+
+    ## XXX figure out how to capture celery tasks kicked off in a subprocess???
```
Comment (arikfr) [inline tests/test_cli.py]: I usually mock the method that submits the task to verify it was called.

File: manage.py (hunk 1 above)
Comment (washort) [inline manage.py]: Oops, that's when I was trying to add a `shell` command and not noticing flask already provided one.

File: tests/test_cli.py (hunk 2 above)
Comment (washort) [inline tests/test_cli.py]: Yes, the problem is that the click testing stuff runs the code in a subprocess, so you can't directly mock. I think I might split the actual work into a separate function and test it separately.

Comment (washort) [inline tests/test_cli.py]: I was wrong, it just captures stdio. Test fixed.

**Pull request merged**